app/src/main/java/org/smartregister/sample/fp/app/FPApplication.kt

```diff
@@ -0,0 +1,60 @@
+package org.smartregister.sample.fp.app
+
+import com.evernote.android.job.JobManager
+import org.smartregister.BuildConfig
+import org.smartregister.Context
+import org.smartregister.CoreLibrary
+import org.smartregister.configurableviews.ConfigurableViewsLibrary
+import org.smartregister.repository.Repository
+import org.smartregister.sample.fp.job.FPJobCreator
+import org.smartregister.sample.fp.repository.FPRepository
+import org.smartregister.util.Log
+import org.smartregister.view.activity.DrishtiApplication
+
+open class FPApplication : DrishtiApplication() {
+
+    private var mPassword: String? = null
+
+    override fun onCreate() {
+        super.onCreate()
+
+        mInstance = this
+        context = Context.getInstance()
+        context.updateApplicationContext(applicationContext)
+
+        //Initialize Modules
+        CoreLibrary.init(context, FPConfiguration(), BuildConfig.BUILD_TIMESTAMP)
+        ConfigurableViewsLibrary.init(context)
+
+        //init Job Manager
+        JobManager.create(this).addJobCreator(FPJobCreator())
+    }
+
+
+    override fun getRepository(): Repository? {
+        try {
+            if (repository == null) {
+                repository = FPRepository(getInstance()?.applicationContext, context)
+            }
+        } catch (e: UnsatisfiedLinkError) {
+            Log.logError("Error on getRepository: $e")
```

Comment:
Switch any log operations done using Log to `Timber`. Timber enables us to easily define what should not be logged in production and even what can be sent to the server by using trees. By default the default provided trees will disable logging in production for security & performance of certain tags

The OpenSRP Timber tree also enables us to send certain logs to the crashlytics/analytics provider